subsys/net/lib/mqtt_sn/mqtt_sn.c

```diff
@@ -725,7 +746,19 @@ static int process_topics(struct mqtt_sn_client *client, int64_t *next_cycle)
 		/* Check if action is due */
 		if (next_attempt <= now) {
 			switch (topic->state) {
+			case MQTT_SN_TOPIC_STATE_SUBSCRIBE:
+				if (subscribing) {
+					/*
+					 * Only one topic can be subscribing or unsubscribing
+					 * at the same time
+					 */
+					break;
+				}
+				topic->state = MQTT_SN_TOPIC_STATE_SUBSCRIBING;
+				LOG_INF("Topic subscription now in progress");
+				/* fallthrough */
```

Comment:
The fallthrough should be marked as
```
__fallthrough;
```

Reply:
fixed

Reply:
ooc, I assume that's a hint for the compiler. Why does it not complain about the `__fallthrough` missing?

Reply:
The compiler will accept also `/* fallthrough */` as zephyr code used that earlier (and there seems to be still code that uses the old way), but at some point the code base was changed to use `__fallthrough` which is the preferred way.